### ISSUE TITLE
`reportUnknownMemberType` false negative in CLI when method has unknown return type

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#6633]
**Describe the bug**
in the code below, the `reportUnknownMemberType` error is only reported in the language server in vscode, but not when running the pyright command

**Code or Screenshots**
```python
class Foo:
    def foo(self, asdf):
        return asdf


foo = Foo()

print(foo.foo(1)) # error: reportUnknownMemberType
```
```
> pyright .\asdf.py
0 errors, 0 warnings, 0 informations
```
![image](https://github.com/microsoft/pyright/assets/57028336/f89848f5-d2b0-4f89-b97d-0167f9ff0537)

**VS Code extension or command-line**
pylance: 2023.11.10
pyright pyi package: 1.1.338

as far as i can tell, this version of pylance is using the same version of pyright

**Comment by erictraut** [comment]
The `reportUnknownMemberType` check isn't enabled by default. You need to enable it. If you want it to be enabled in the CLI, you need to do so in a configuration file. Refer to [this documentation](https://microsoft.github.io/pyright/#/configuration).

**Comment by DetachHead** [comment]
@erictraut i forgot too include it but i just edited the issue too clarify that i do have that option enabled

**Comment by erictraut** [comment]
Please post the contents of your `pyrightconfig.json` or `pyprotject.toml` file.

**Comment by erictraut** [comment]
The code you've posted above does not generate any `reportUnknownMemberType` errors.

Code sample in [pyright playground](https://pyright-play.net/?code=MQAgDgngTglg5gCwC4C4RQKZgPZSQVQDsBrQ7Ad0IFkMBbAIwygBUIwMBeJKAVwwCh%2BAYwA2AQwDOEkADFs2FPxDKQAEwwAzEBvkAKCRhEaANCEmqNASkUrb6DEh5RCZiRcE7sIDrL2XBYLCESLqeAHSeugCMlv5AA)

**Comment by erictraut** [comment]
This behavior changed in pyright 1.1.338. The version of pylance that you're using is based on an older version.

**Comment by erictraut** [comment]
Refer to [this documentation](https://github.com/microsoft/pylance-release/blob/main/USING_WITH_PYRIGHT.md).

**Comment by DetachHead** [comment]
thanks. however that means the version of pyright used in my project has to be tied to whatever the latest version of pylance is, which i can't pin with a lockfile like i can with other dependencies. so if i'm working on a branch that hasn't had its dependencies updated in a while i could see new type errors that aren't caused by any of the changes in that branch seemingly appear out of nowhere, which is quite confusing.

i've raised https://github.com/microsoft/pylance-release/issues/5201 for this

**Comment by erictraut** [comment]
Pylance is architected in a way that doesn't allow for what you're proposing. It's effectively a fork of pyright. You'll need to find a different solution.
